auto-send-to/src/main.ts

```diff
@@ -0,0 +1,86 @@
+import { Metadata } from '@devvit/protos';
+import { Devvit, getSetting } from '@devvit/public-api';
+
+Devvit.use(Devvit.Types.HTTP);
+
+const settingOptions: any = [
+  {
+    defaultValue: 'posts',
+    type: 'select',
+    name: 'listenerObject',
+    label: 'Objects to Send On',
+    options: [
+      { label: 'Posts', value: 'posts' },
+      { label: 'Comments', value: 'comments' },
+      { label: 'Posts and Comments', value: 'both' },
+    ],
+  },
+  {
+    type: 'string',
+    name: 'webhook',
+    label: 'Webhook URL',
+  },
+  {
+    type: 'string',
+    name: 'flairId',
+    label: 'Only send when post has this flair ID (optional, will not send comments too)'
+  }
+];
+
+Devvit.addSettings(settingOptions);
+
+Devvit.addTrigger({
+  events: [Devvit.Trigger.PostSubmit, Devvit.Trigger.CommentSubmit],
+  handler: sendContentToWebhook,
+});
+
+async function sendContentToWebhook(event: Devvit.MultiTriggerEvent, metadata?: Metadata) {
+  const webhook = await getSetting('webhook', metadata) as string
+  const listenerObject = await getSetting('listenerObject', metadata) as string
+  const flairId = await getSetting('flairId', metadata) as string
+  if (!webhook) {
+    throw new Error('No webhook URL provided');
+  }
+
+  if (
+    (listenerObject === 'posts' && event.type === Devvit.Trigger.CommentSubmit) ||
+    (listenerObject === 'comments' && event.type === Devvit.Trigger.PostSubmit)
+  ) {
+    return;
+  }
+
+  if (flairId) {    
+    if ((event.type === Devvit.Trigger.CommentSubmit || event.type === Devvit.Trigger.PostSubmit)) {
+      const postFlairId = event.event?.post?.linkFlair?.templateId ?? '' as string;
```

Comment:
:no_bell: Do we need the type assertion? It looks like we coalesce down to string if no value.